swiftwinrt/code_writers.h

```diff
@@ -470,43 +470,43 @@ bind<write_abi_args>(function));
     static void write_ireference_init_extension(writer& w, generic_inst const& type)
     {
         if (!is_winrt_ireference(type)) return;
+        auto generic_param = type.generic_params()[0];
+        w.add_depends(*generic_param);
+
+        auto impl_names = w.push_impl_names(true);
+
+        auto c_name = is_struct(*generic_param) ?
+            generic_param->mangled_name() :
+            generic_param->cpp_abi_name();
 
-        auto format = R"(internal extension % {
-    init?(ref: UnsafeMutablePointer<%>?) {
-        guard let val = ref else { return nil }
+        w.write(R"(internal struct %: ReferenceBridge {
```

Comment:
```suggestion
        w.write(R"(internal enum%: ReferenceBridge {
```
(if it's not meant to be constructed)